### ISSUE TITLE
[FEATURE-REQUEST] Expo support?

### ISSUE THREAD

**Issue by jay-jlm** [daily-co/react-native-daily-js#30]
A good amount of RN projects these days rely on Expo, so I thought it would be useful if some information was added to the ReadME file about the status of Expo support.
I'd love to see this library go the same route as many others have gone in supporting Expo's new "dev client" workflow, which allows developers the installation of custom native code while staying in a managed workflow that does not require local builds and local "ios" and "android" folders.
More information on this topic here: https://docs.expo.dev/guides/config-plugins/

**Comment by filipi87** [comment]
Hi @jay-jlm 

Thanks a lot for the heads up, you are right, one of our main limitations for supporting expo, was because we were unable to support webrtc. There was even a [feature request for that on expo](https://expo.canny.io/feature-requests/p/webrtc),
which is now mentioning this same alternative of config plugin to solve that issue.

I will share this information with our product team, and ask them to add this feature to our backlog. I will let you know as soon as I have some updates.

**Comment by jay-jlm** [comment]
https://www.memecreator.org/static/images/memes/4712649.jpg

**Comment by rodolphoasb** [comment]
Hey, @filipi87! How are you doing? Do you know some projects using Daily and Expo React Native in production? 

Everything is working fine, but after running the [installation steps](https://www.npmjs.com/package/@daily-co/react-native-daily-js) for ios I always end up stuck with this screen:

<img width="416" alt="image" src="https://user-images.githubusercontent.com/8127858/215598874-a76deadb-f456-421f-b091-f9ea5ea27f80.png">

I'm using Expo EAS build with these commands:

`eas build -p ios --profile development --local`
`expo start --dev-client`

Also tried to run it with:

`expo run:ios`

There's no log of errors.

EDIT 00:

I tried to use this config plugin by Expo: https://github.com/expo/config-plugins/tree/main/packages/react-native-webrtc as described here: https://expo.canny.io/feature-requests/p/webrtc. But haven't had any luck with it!

**Comment by filipi87** [comment]
Hi @rodolphoasb ,

Unfortunately, we have not yet added official support for expo in our library. In order to make `react-native-daily-js` work with **Expo**, we would probably:

- Create an [Expo config plugin](https://docs.expo.dev/guides/config-plugins/) for `react-native-daily-js`, which should respect the requirements described in our page of [react-native-daily-js](https://github.com/daily-co/react-native-daily-js): 
  - iOS: Deployment target >= 11.0
  - Android: minSdkVersion >= 24
   - For iOS: add additional configs on `Info.plist`
     ```
     <key>UIBackgroundModes</key>
     <array>
        <string>voip</string>
     </array
     ```
   - For Android: add additional configs on `AndroidManifest.xml`
      ```
         <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
         <service android:name="com.daily.reactlibrary.DailyOngoingMeetingForegroundService"/>
       ```

The other requirements, for both iOS and Android, like the other permissions, looks like It is already contemplated on the `react-native-webrtc` [plugin](https://github.com/expo/config-plugins/tree/main/packages/react-native-webrtc).

But even that `react-native-webrtc` would probably need a couple of changes, since we don't use the version from the upstream, we use our [fork](https://github.com/daily-co/react-native-webrtc), which is a fork from version 1.94.1.

So, looking at the [react-native-webrtc config plugin]( https://github.com/expo/config-plugins/blob/main/packages/react-native-webrtc/ ), when they mention how to install, instead of install the version from upstream, we would need to install the version from Daily:

`npx expo install @daily-co/react-native-webrtc @config-plugins/react-native-webrtc`

Other point that may be critical, is that they mention this table of compatibility:
| `expo` | `react-native-webrtc` | `@config-plugins/react-native-webrtc` |
| ------ | --------------------- | ------------------------------------- |
| 47.0.0 | 1.106.1               | 5.0.0                                 |
| 46.0.0 | 1.100.0               | 4.0.0                                 |
| 45.0.0 | 1.100.0               | 3.0.0                                 |
| 44.0.0 | 1.92.2                | 2.0.0                                 |
| 43.0.0 | 1.92.2                | 1.0.0                                 |

So, not sure what version of expo would be compatible with the version `1.94.1`, which is the one currently used by Daily. I would probably try Expo `44.0.0` to start or expo `45.0.0`.

In summary, to add support for Expo, we would probably need:
- Make some adjustments on `react-native-webrtc plugin`, in order to make It work with our fork;
- Create a config plugin for `react-native-daily-js`;

We have an open task to do that, but It has not been prioritized yet.

**Comment by filipi87** [comment]
Hi, we have prioritized the task to add support for Expo.

We are on the final steps of code review to release a config plugin that will allow using Daily together with Expo.

I will let you know as soon as It is available.

Thanks a lot for reporting this.

cc: @jay-jlm , @rodolphoasb

**Comment by filipi87** [comment]
Hi, I am glad to share that we have added the Expo Support to Daily.

In order to use Daily together with Expo, you should follow the instructions shared in our plugin [@daily-co/config-plugin-rn-daily-js](https://www.npmjs.com/package/@daily-co/config-plugin-rn-daily-js).

Or you can take a look at our [demo](https://github.com/daily-demos/daily-expo-demo/), a simple app showcasing `react-native-daily-js` working together with Expo.

Let me know if you have any questions or face any issue while using It.
